src/pages/home/ReportScreen.js

```diff
@@ -111,9 +112,14 @@ class ReportScreen extends React.Component {
         this.chatWithAccountManager = this.chatWithAccountManager.bind(this);
         this.dismissBanner = this.dismissBanner.bind(this);
 
+        // We need sortedAndFilteredReportActions to be set before this.state is initialized
+        // because the function to calculate the newMarkerReportActionID (ReportActionsView) uses the sorted report actions
+        const sortedAndFilteredReportActions = ReportActionsUtils.getSortedReportActionsForDisplay(props.reportActions);
+
         this.state = {
             skeletonViewContainerHeight: reportActionsListViewHeight,
             isBannerVisible: true,
+            sortedAndFilteredReportActions,
```

Comment:
Is this just derived state? It feels like an anti-pattern to store this in state since we can derive it from props. What's the reason for doing this?

Reply:
I see your pointm, I think the idea was to not having to call this function on two places [here](https://github.com/Expensify/App/pull/15207/files#diff-dd4bfa50713397aca8cb40145317936ab0affe19abc2a9c24d6c8849ed75dc9bR287) and [here](https://github.com/Expensify/App/pull/15207/files#diff-dd4bfa50713397aca8cb40145317936ab0affe19abc2a9c24d6c8849ed75dc9bR298)

Reply:
Also, If the `reportActions` prop changes, we need to re-sort and re-render the component. Even though the sorted data can be derived from props, we still need to store it in state to keep it up-to-date and avoid issues like User B sending a message to User A while User A is in the report, the new message won't show up

Reply:
I get that, but won't the change in props trigger a re-render of the component anyways (unless we explicitly disabled that in `shouldComponentUpdate`)? Which means we can re-sort the `reportActions` at render time.

Reply:
>I think the idea was to not having to call this function on two places [here](https://github.com/Expensify/App/pull/15207/files#diff-dd4bfa50713397aca8cb40145317936ab0affe19abc2a9c24d6c8849ed75dc9bR287) and [here](https://github.com/Expensify/App/pull/15207/files#diff-dd4bfa50713397aca8cb40145317936ab0affe19abc2a9c24d6c8849ed75dc9bR298)

We could create a variable for it in the render() method

Reply:
We discussed it [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1677778414008709) and determined that we should do the filter/sort in `render()` since performance shouldn't be an issue.

Reply:
Also, as a side note Onyx does support custom props via the `selector` key, as we do [here](https://github.com/Expensify/App/blob/4512136a17062147ce0e38b7029d7769cd903a34/src/pages/home/sidebar/SidebarLinks.js#L290)

Reply:
@luacmartins Awesome, on my way, thanks for the tips, would be good to add them in `contributingGuides/STYLE.md`

Reply:
@luacmartins @mountiny  Went through the slack thread and wanna grab some data, is true that there isn't much difference sortingAndFiltering on render, what I could get is a 10% improvement using the derived state.

<img width="1273" alt="Screen Shot 2023-03-03 at 12 25 25" src="https://user-images.githubusercontent.com/1676818/222708865-c472eca5-ffbd-45ae-8ffc-e2532b172d9f.png">

Average:
Props: ~300ms
State: ~268ms

Doing Remounting x5 times, and forcing update x15 times

Reply:
I just saw that I can add a selector to Onyx, I runed the same tests and got almost the same numbers as state:
<img width="984" alt="Screen Shot 2023-03-03 at 12 40 03" src="https://user-images.githubusercontent.com/1676818/222711363-ea6f00e5-d3e6-461f-99c4-9c44a9cb78f9.png">

The green one

Average:
Onyx with selector: ~260ms

Doing Remounting x5 times, and forcing update x15 times

I would recommend in this cases using the selector from Onyx instead of sorting over the render

All this cases were on a Physical device, OnePlus6 Android 11

Reply:
Thanks! Lets use the onyx selector

Reply:
Sounds good. Let me know when the changes are ready for review

Reply:
@luacmartins uploaded